pyo3-macros-backend/src/pyclass.rs

```diff
@@ -526,12 +526,22 @@ impl<'a> PyClassComplexEnum<'a> {
                             options,
                         })
                     }
-                    Fields::Unnamed(_) => {
-                        bail_spanned!(variant.span() => format!(
-                        "Tuple variant `{ident}` is not yet supported in a complex enum\n\
-                        = help: change to a struct variant with named fields: `{ident} {{ /* fields */ }}`\n\
-                        = note: the enum is complex because of non-unit variant `{witness}`",
-                        ident=ident, witness=witness))
+                    Fields::Unnamed(types) => {
+                        let fields = types
+                            .unnamed
+                            .iter()
+                            .enumerate()
+                            .map(|(_i, field)| PyClassEnumVariantUnnamedField {
+                                ty: &field.ty,
+                                span: field.span(),
+                            })
```

Comment:
Clippy is reporting the needless `enumerate`

```suggestion
                            .map(|field| PyClassEnumVariantUnnamedField {
                                ty: &field.ty,
                                span: field.span(),
                            })
```